### ISSUE TITLE
Variable declaration via `list` inside `foreach` not recognized

### ISSUE THREAD

**Issue by gmazzap** [sirbrillig/phpcs-variable-analysis#33]
Following is [perfectly valid PHP 5.5+ code](https://3v4l.org/h5hYc):

```php
$data = [
    ['foo', 'Foo'],
    ['bar', 'Bar'],
];

foreach ($data as list($name, $label)) {
    printf('<div id="%s">%s</div>', $name, $label);
}
```

However, it raises the following warnings:

```
------------------------------------------
WARNING | Variable $name is undefined.
WARNING | Variable $label is undefined.
------------------------------------------
```

**Comment by sirbrillig** [comment]
Thanks for the report!

**Comment by gmazzap** [comment]
I think the fix actually broke _used_ variables check inside `foreach`.

In my code I have:

```php
foreach ($callbacks as $callback) {
  $callback();
 }
```

Which causes a:

```
WARNING | Unused variable $callback.
```

But I could notify that there are a lot of things that now raise unused variables false positive for variable declared by `foreach`.

**Comment by gmazzap** [comment]
Not sure if I should I open another issue, or maybe you could re-open this.

**Comment by sirbrillig** [comment]
D'oh! I guess these unit tests are not good enough to prevent regressions. 😭 Well, we'll make them better.

**Comment by sirbrillig** [comment]
Sorry about that. There's still one bug I'm tracking in #37 but 2.0.7 should fix the major bug.
